simulator/risc_v/t/unit_test.cpp

```diff
@@ -429,6 +432,24 @@ TEST_RV64_RR_OP( 3, bset, 0xFFFFFFFFFFFFFFF4, 0xFFFFFFFFFFFFFFF0, 0x2)
 TEST_RV64_RR_OP( 4, bset, 0xAAAAAAAAAAAAAAA2, 0xAAAAAAAAAAAAAAA0, 0xFFFFFFFFFFFFFF01) // overflow test
 TEST_RV64_RR_OP( 5, bset, 0x1111111111111100, 0x1111111111111000, 0xFFFFFFFFFFFFFF08) // overflow test
 
+TEST_RV32_RR_OP( 1, sext_b, 0x7FFFFF90, 0x00000090, 0) // check for preservence of sign for positive
+TEST_RV32_RR_OP( 2, sext_b, 0x7FFFFF90, 0x7FFFFF90, 0) //
```

Comment:
Why the MSB (bit 31) is still zero?

Reply:
I thought that the MSB is viewed as a bit which is responsible for the sign and in sign extension the value has to remain the same sign

Reply:
Spec mentions _the most-significant bit in the byte (i.e., bit 7)_